projectq/backends/_sim/_pysim.py

```diff
@@ -20,6 +20,11 @@
 
 import random
 import numpy as _np
+import os
+
+_USE_REFCHECK = True
+if 'TRAVIS' in os.environ:
+    _USE_REFCHECK = False
```

Comment:
Why do you set refcheck `false` for travis?

Reply:
This is because of some errors that seem to happen only when we run the coverage tests with pytest. I am assuming this happens because the coverage tests injects some code somewhere to properly keep track of the code coverage status and that may interfere with some of numpy reference checking mechanisms.

```
ValueError: cannot resize an array references or is referenced
by another array in this way.  Use the resize function
```

However, I have not been able to find any helpful documentation so far to properly explain that behaviour. Plus I can't seem to reliably reproduce this behaviour on my machines at the moment.

As far as I can remember, running the tests without the coverage plugin does not lead to the same error, so I am reluctant to entirely disable the feature by passing `refcheck=False` to `resize(...)` since that might still help catch some unwanted behaviour.

Reply:
One other way to solve that issue would be using the free function `numpy.resize(...)` instead of the array method, but that would probably induce more allocation/deallocations of memory leading to some worse performance.

Reply:
ok, makes sense then.